lib/facade_converter.ts

```diff
@@ -92,19 +95,34 @@ export class FacadeConverter extends base.TranspilerBase {
    * Searches for type references that require extra imports and emits the imports as necessary.
    */
   emitExtraImports(sourceFile: ts.SourceFile) {
-    var hasXmlHttpRequest = (n: ts.Node): boolean => {
-      if (n.kind === ts.SyntaxKind.TypeReference) {
-        var type = base.ident((<ts.TypeReferenceNode>n).typeName);
-        if (type === 'XMLHttpRequest') {
-          return true;
+    var libraries = <ts.Map<string>>{
+      "XMLHttpRequest": "dart:html",
+      "Uint8Array": "dart:typed_arrays",
+      "ArrayBuffer": "dart:typed_arrays"
+    };
+    var emitted: Set = {};
+    this.emitImports(sourceFile, libraries, emitted);
```

Comment:
I find not passing `sourceFile` in this call and then doing the cast below quite confusing. Could you make the parameter non-optional and just pass `sourceFile` twice here?